lib/tests/formulae.rb

```diff
@@ -82,39 +103,27 @@ def detect_formulae!(args:)
             "#{@argument} is not detected from GitHub Actions or a formula name!"
         end
 
-        # Use GitHub Actions variables for pull request jobs.
-        if ENV["GITHUB_BASE_REF"].present? && ENV["GITHUB_SHA"].present?
-          diff_start_sha1 = rev_parse("origin/#{ENV["GITHUB_BASE_REF"]}")
-          if diff_start_sha1.blank?
-            test git, "-C", repository, "fetch", "--depth=1",
-                 "origin", "+refs/heads/#{ENV["GITHUB_BASE_REF"]}",
-                 args: args
-            diff_start_sha1 = rev_parse("origin/#{ENV["GITHUB_BASE_REF"]}")
+        if ENV["GITHUB_REPOSITORY"].present?
+          # We're using GitHub Actions. Check the repo matches the tap we are using.
+          if tap&.full_name&.downcase == ENV["GITHUB_REPOSITORY"].downcase
+            diff_start_sha1, diff_end_sha1 = github_actions_revisions
           end
-          diff_end_sha1 = ENV["GITHUB_SHA"]
-        # Use GitHub Actions variables for branch jobs.
-        elsif ENV["GITHUB_SHA"].present?
-          diff_end_sha1 = diff_start_sha1 = ENV["GITHUB_SHA"]
-        # Otherwise just use the current SHA-1 (which may be overriden later)
-        else
-          if ENV["CI"] || ENV["GITHUB_ACTIONS"]
-            onoe <<~EOS
-              No known CI provider detected! If you are using GitHub Actions then we cannot find the expected  environment variables! Check you have e.g. exported them to a Docker container.
-            EOS
-          end
-          diff_end_sha1 = diff_start_sha1 = current_sha1
+        elsif ENV["CI"]
+          onoe <<~EOS
+            No known CI provider detected! If you are using GitHub Actions then we cannot find the expected environment variables! Check you have e.g. exported them to a Docker container.
+          EOS
         end
 
+        diff_start_sha1 = current_sha1 if diff_start_sha1.blank?
```

Comment:
Why is this moved? I don't think it should be here because we don't want to calculate the merge base based on these.

Reply:
The behaviour isn't really too different than before - if `GITHUB_SHA` wasn't present then it already did `diff_start_sha1 = diff_end_sha1 = current_sha1`. This was just a replacement of that line but perhaps it was just better to just remove the line and leave this below.

Reply:
Yeh, would be ideal to scope that to only if `GITHUB_SHA` isn't present.

Reply:
I mean - does it even need to apply when `GITHUB_SHA` isn't present (i.e. not GitHub Actions)? I don't think we need to `merge-base` in that scenario either.